Sources/SwiftDocC/Infrastructure/Link Resolution/PathHierarchy.swift

```diff
@@ -207,9 +207,17 @@ struct PathHierarchy {
                         parent.children[components.first!] == nil,
                         "Shouldn't create a new sparse node when symbol node already exist. This is an indication that a symbol is missing a relationship."
                     )
+                    guard knownDisambiguatedPathComponents != nil else {
```

Comment:
We don't _need_ to do this but it's a small optimization to avoid doing unnecessary work. Before we always parsed each placeholder node's path component for disambiguation but now we only do it if we know that the path hierarchy has been provided some custom path components (which could include disambiguation that we need to parse)

Reply:
I added a brief comment that talks about this in https://github.com/apple/swift-docc/pull/828/commits/50cc145e9db59a89a3b35b595294eb613fce4736